### PR TITLE
switch Microsoft OpenJDK to use static URL's

### PR DESCRIPTION
**Pull request by gdams**
fixes: https://github.com/actions/setup-java/issues/296

**Description:**
Describe your changes.

**Related issue:**
Add link to the related issue.

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Comment by joschi** [comment]
@gdams Referring to this code comment:
> We will need Microsoft to add an endpoint where we can query for versions.

Is there any URL which can be used to retrieve metadata about (all) the Microsoft OpenJDK builds, maybe even an API similar to the [Adoptium API](https://api.adoptium.net/q/swagger-ui/)?

**Comment by brunoborges** [comment]
@joschi we don't have that ready yet.

File: dist/setup/index.js [hunk 1]
```diff
@@ -13913,20 +13913,18 @@ class MicrosoftDistributions extends base_installer_1.JavaBase {
...
                 {
-                    version: [17, 0, 1, 12, 1]
+                    version: [17]
                 },
                 {
-                    version: [16, 0, 2, 7, 1]
+                    version: [16]
                 }
             ];
             // M1 is only supported for Java 16 & 17
```
Comment (brunoborges) [inline dist/setup/index.js:13924]: @gdams we now provide JDK 11 for M1 as well. I think this block can be simplified.

**Review by brunoborges — CHANGES_REQUESTED** [review]

**Comment by dmitry-shibanov** [comment]
Hello @gdams. Could you please run these `npm run build && npm run format` commands.

**Comment by dmitry-shibanov** [comment]
Hello @gdams, just a gentle ping.

**Review by Skiba36 — APPROVED** [review]

**Comment by karianna** [comment]
@gdams - needs a rebase

**Comment by dmitry-shibanov** [comment]
Hello everyone. I'm going to close the pull request because we added manifest with versions for Microsoft OpenJDK. Now we can add new versions to the manifest file without releasing new version of the action.

**Pull request closed without merging**